### DISCUSSION TITLE
Release 23.1

### DISCUSSION THREAD

**Discussion by commi235** [flowsforapex/apex-flowsforapex#672] [General] [answered]
## Changes in this Release ##

- Adds Due Dates and Priorities to bpmn:processes and to bpmn:userTasks, allowing declarative tracking of scheduling, priority and due dates.
- Adds Task Assignment to bpmn:userTasks, allowing easier declarative assignment of tasks to potential users and groups.
- Adds REST-enabling of the Flows for APEX API 
- Major upgrade to the BPMN Modeler, with new, modern-looking properties panel.
- Enhanced support for modeling and viewing of collapsed subProcesses that can be easily expanded / collapsed.
- Adds support for BPMN Child Lane sets, allowing nesting of lane sets inside parent lanes.
- Adds explicit definition of relationship (or not) between Lanes and APEX roles/groups.
- Adds Timestamp with Time Zone datatype option for Process Variables.
- Adds binding of Process Variables into PL/SQL scripts, variable expressions, and element settings.
- Adds support for binding non-varchar2 typed (i.e., number, date, timestamp with time zone) process variables. 
- Allows process variable substitutions to be used in all of the parameters of an APEX Page task.
- Adds 'is Startable' information to Processes, along with users and groups that can start a process.
- Allows BPMN diagrams to contain Groups and attached Annotations.
- Adds Automation to create daily, month-to-date, monthly, and quarterly statistics on process and step performance.
- Enhances Flows for APEX Application dashboard to display process and step performance, errors, and waiting time statistics.
- Adds auditing / logging of changes made to process diagrams, to complete the existing logging capabilities
- Adds the ability to create a summary document, as a JSON document, describing all actions taken to run a process instance.
- Adds an Automation creating archive summaries for all recently completed process instances into an archive table or OCI Object Storage.
- Adds an automation for automated purging of old logging data and statistics summaries.
- Allows an administrator to easily change the timer polling frequency, or temporarily enable / disable timers on non-production systems.
- Experimental Feature - Adds MessageFlow between sendTasks, receiveTasks, Intermediate Message Catch and Throw events. Phase 1 of more...
- Adds support for additional languages Italian, Simplified Chinese, Traditional Chinese, Korean.
- Required APEX version increased to APEX v20.2

### 1.  Fresh Install ###

See [installation instructions](https://flowsforapex.org/latest/installation/)
 
### 2. Upgrade from Flows for APEX v22.1 ###

See [Migration Instructions](https://flowsforapex.org/latest/migration/)

### 3. Plugin for APEX Approval Tasks Integration (APEX 22.1+) ###

1. Download the file `f4a_approval_return_result_plugin.sql`.
2. Import the plugin into your application.
3. More information can be found in our documentation.

### 4. Enabling the REST API

Use the steps as documented in the [installation instructions](https://flowsforapex.org/latest/installation/).
We also include a Postman Collection and the OpenAPI specification in the release assets.
- `FlowsForAPEX_v23.1_Postman_Collection.zip`
- `FlowsForAPEX_v23.1_openapi_3_0_0.yml`

### 5. Combined Task List for APEX Approval Tasks and Flows User Tasks

If you are running on an APEX release which support Approval Tasks you might want to run `FlowsForAPEX_v23.1_Combined_Task_List.sql` as the Flows schema from the assets below.
This view allows you to have a combined view of Flows & APEX Tasks.
You can find more information in our [documentation](https://flowsforapex.org/latest/tasklists/).

<hr /><em>This discussion was created from the release <a href='https://github.com/flowsforapex/apex-flowsforapex/releases/tag/v23.1'>Release 23.1</a>.</em>

**Comment by lipe-8haus** [comment]
Where can I find the file "no_sqlnet_migration.zip" for this release?
Thanks!

**Reply by commi235** [reply]
We are currently producing this file.
Should be available until end of this week.

**Reply by commi235** [reply, marked as answer]
File has been uploaded to the release
